WordPress/gutenberg
# [RNMobile] Introduce a common API to style Button from both web and mobile

## Description
This is an exploration of x-platform style system. https://github.com/wordpress-mobile/gutenberg-mobile/issues/1411
This is the second iteration on https://github.com/wordpress-mobile/gutenberg-mobile/issues/1386

Gutenberg-mobile PR - https://github.com/wordpress-mobile/gutenberg-mobile/pull/1684

In this PR I used `styled-system` with `emotion` to create a primitive Button component that could be used in web and mobile and could be styled via props. That PR contains the mobile part.

- I added the `button` primitive component (`TouchableOpacity`) that can be used with `styled-system`
- I added the theme provider and theme values (they can be a bit outdated since I took them from my previous PR https://github.com/WordPress/gutenberg/pull/17614 )
- I used created primitive in `@components/Button`
- I added `isSmall`, `isLarge` and `isPrimary` props and set button style related to them (just to demonstrate)
- I used Button primitive in breadcrumbs
- I set `isLarge` and `isPrimary` to `@components/Button` in `ButtonBlockAppender` component to demonstrate they work :)

## How has this been tested?
- Run `gutenberg-mobile` app
- All buttons should look the same as before
- Check if breadcrumbs are rendered properly
- Check if block appender has `blue` background and additional padding (because of `isLarge` and `isPrimary`)

## Screenshots 
### without `isLarge` and `isPrimary`
![Simulator Screen Shot - iPhone X - 2019-12-16 at 12 06 35](https://user-images.githubusercontent.com/16336501/71001923-8be39d80-20de-11ea-8ead-3ff2cd097156.png)

### with `isLarge` and `isPrimary`
![Simulator Screen Shot - iPhone X - 2019-12-16 at 13 52 30](https://user-images.githubusercontent.com/16336501/71001973-a61d7b80-20de-11ea-9941-b9b4f4a4fc3e.png)

## Types of changes
Experimenting with `emotion` and `styled-system`
